android/src/main/java/com/brentvatne/exoplayer/FullScreenPlayerView.kt

```diff
@@ -0,0 +1,124 @@
+package com.brentvatne.exoplayer
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import android.view.ViewGroup
+import android.view.WindowManager
+import android.widget.FrameLayout
+import android.widget.ImageButton
+import androidx.activity.OnBackPressedCallback
+import androidx.media3.ui.LegacyPlayerControlView
+import com.brentvatne.common.toolbox.DebugLog
+import java.lang.ref.WeakReference
+
+@SuppressLint("PrivateResource")
+class FullScreenPlayerView(
+    context: Context,
+    private val exoPlayerView: ExoPlayerView,
+    private val reactExoplayerView: ReactExoplayerView,
+    private val playerControlView: LegacyPlayerControlView?,
+    private val onBackPressedCallback: OnBackPressedCallback
+) : Dialog(context, android.R.style.Theme_Black_NoTitleBar_Fullscreen) {
+
+    private var parent: ViewGroup? = null
+    private val containerView = FrameLayout(context)
+    private val mKeepScreenOnHandler = Handler(Looper.getMainLooper())
+    private val mKeepScreenOnUpdater = KeepScreenOnUpdater(this)
+
+    private class KeepScreenOnUpdater(fullScreenPlayerView: FullScreenPlayerView) : Runnable {
+        private val mFullscreenPlayer = WeakReference(fullScreenPlayerView)
+
+        override fun run() {
+            try {
+                val fullscreenVideoPlayer = mFullscreenPlayer.get()
+                if (fullscreenVideoPlayer != null) {
+                    val window = fullscreenVideoPlayer.window
+                    if (window != null) {
+                        val isPlaying = fullscreenVideoPlayer.exoPlayerView.isPlaying
+                        if (isPlaying) {
+                            window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+                        } else {
+                            window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+                        }
+                    }
+                    fullscreenVideoPlayer.mKeepScreenOnHandler.postDelayed(this, UPDATE_KEEP_SCREEN_ON_FLAG_MS)
+                }
+            } catch (ex: Exception) {
+                DebugLog.e("ExoPlayer Exception", "Failed to flag FLAG_KEEP_SCREEN_ON on fullscreen.")
+                DebugLog.e("ExoPlayer Exception", ex.toString())
+            }
+        }
+
+        companion object {
+            private const val UPDATE_KEEP_SCREEN_ON_FLAG_MS = 200L
+        }
+    }
+
+    init {
+        setContentView(containerView, generateDefaultLayoutParams())
+    }
+
+    override fun onBackPressed() {
```

Comment:
nit;
It's not point of this PR, onBackPressed was deprecated🥲 so, it would be nice to change it.